connectionclass/src/main/java/com/facebook/network/connectionclass/DeviceBandwidthSampler.java

```diff
@@ -113,7 +113,7 @@ public void handleMessage(Message msg) {
      * Method for polling for the change in total bytes since last update and
      * adding it to the BandwidthManager.
      */
-    private void addSample() {
+    private void addSample(boolean cleanPreviousReadBytes) {
```

Comment:
I really don't like boolean params like this.  I feel it makes the calling code hard to read.  Lets add a function called addFinalSample() which calls addSample() then calls QTagParser.resetPreviousBytes();